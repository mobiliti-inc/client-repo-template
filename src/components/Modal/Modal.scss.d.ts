// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'fades-in-out': string;
  'header': string;
  'header-no-icon': string;
  'hide': string;
  'modal': string;
  'modal--dark-gray': string;
  'modal--gray': string;
  'modal--white': string;
  'modal-body': string;
  'modal-container': string;
  'modal-content': string;
  'modal-footer': string;
  'modal-icon': string;
  'modal-icon-close': string;
  'modal-plain': string;
  'modal-sub-title': string;
  'modal-title': string;
  'no-pointer': string;
  'plain-modal-header': string;
  'scroll': string;
  'visible': string;
}
declare var cssExports: CssExports;
export = cssExports;
