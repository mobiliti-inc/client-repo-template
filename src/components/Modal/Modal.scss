@import "../../styles/_variables.scss";

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 200;
  cursor: pointer;
  transition: all 0.5s ease;
}

.modal--white,
.modal--gray,
.modal--dark-gray {
  composes: modal
}

.modal--white {
  background: rgba(255, 255, 255, 0.8);
}

.modal--gray {
  background: $blue-overlay
}

.modal--dark-gray {
  background: rgba(41, 41, 41, 0.4);
}

.visible {
  visibility: visible;
  opacity: 1;
  overflow: scroll;
}

.hide {
  visibility: hidden;
  opacity: 0;

}

.no-pointer {
  cursor: auto;
}

.modal-content {
  background-color: white;
  margin: 0 auto;
  max-height: 100%;
  width: 35%;
  box-shadow: 0 20px 40px 0 rgba(116, 39, 39, 0.09);
  transform: translate(0, 75%);
  display: flex;
  flex-direction: column;

  @media (max-width: $half-mobile-width) {
    width: 100%;
    height: 100%;
  }

  @media (max-width: $mobile-width) {
    width: 100%;

  }

  @media (min-width: $half-mobile-width) and (max-width: $mobile-width) {
    width: 100%;
  }

  @media (min-width: $half-mobile-width) and (max-width: $tablet-width) {
    max-width: 60%;
  }
}

.modal-plain {
  transform: translate(0, 62%);

  @media (max-width: $mobile-width) {
    transform: translate(0, 12%);

  }
}


.modal-content.scroll {
  transform: translate(0, 40%);

  @media (max-width: $half-mobile-width) {
    transform: translate(0, 0);
  }
}

.modal-container {
  display: flex;
  flex-direction: column;
  margin: 10%;
  height: 100%;

  @media (max-width: $half-mobile-width) {
    margin: 5%;
  }
}

.header {
  height: 50%
}

.header-no-icon {
  height: 35%;
}

.plain-modal-header {
  display: flex;
  padding-bottom: 10px;
  justify-content: flex-end;
  cursor: pointer;
}

.modal-icon {
  box-sizing: border-box;
  height: 68px;
  width: 68px;
  border: 2px solid $darker-blue;
  border-radius: 50%;
  display: flex;
  justify-content: center;
}

.modal-title {
  color: $darker-blue;
  font-family: $futura;
  font-size: 28px;
  font-weight: bold;
  line-height: 31px;
  margin-bottom: 0;
  margin-top: 10px;
  width: 100%;
}

.modal-sub-title {
  color: $darker-blue;
  font-family: $hanken-sans;
  font-size: 16px;
  line-height: 19px;
  margin: 0;
  padding-top: 5px;
  width: 100%;
  font-weight: normal;
}

.modal-body {
  display: flex;
  flex-direction: column;
  margin-top: 20px;
}

.modal-footer {
  display: block;
  bottom: 64px;
  position: absolute;

  >button {
    margin: 0;
    padding-left: 60px;
    padding-right: 60px;
  }

  @media (max-width: $half-mobile-width) {
    bottom: 50px;
  }
}

.modal-icon-close {
  height: 15px;
  width: 15px;
}