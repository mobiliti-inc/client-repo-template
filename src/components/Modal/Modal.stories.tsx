import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import * as React from 'react';
import { Modal } from '../';

storiesOf('Modal', module)
	.add('default', () =>
		<Modal
			title={text('title', 'Example Title')}
			subTitle={text('subTitle', 'Example Subtitle')}
			buttonText={text('buttonText', 'Continue')}
			visible={boolean('visible', true)}
			onButtonClick={() => { }}
			scrolling={boolean('scrolling', true)}
			plain={boolean('plain', false)}
			overlay={text('overlay', 'white')}
			freezeOverlay={boolean('freezeOverlay', false)}
			noIcon={boolean('noIcon', false)}
			showCloseIcon={boolean("showCloseIcon", false)}
			showButtonLoader={boolean('showButtonLoader', false)}
			onIconClose={() => { }}
			plainHeader={boolean('plainHeader', false)}
			noPointer={boolean('noPointer', false)}
		>
			<h2>Modal Body</h2>
		</Modal >, {
			notes: 'Overlay can be gray, white or dark-gray'
		}
	);
