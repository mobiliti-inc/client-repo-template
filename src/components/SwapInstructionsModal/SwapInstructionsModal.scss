@import '~variables';

.modal {
	border-radius: 5px;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}

.modal-title {
	font-size: 24px;
	font-family: $futura;
	font-weight: bold;
	color: $darker-blue;
	margin-top: 0;
}

.modal-ctn {
	white-space: normal;

	a:last-child {
		>div {
			border: none;
		}
	}
}

.button-container {
	display: flex;
	flex-direction: row;
	margin-top: 20px;

	button {
		width: 40%;

		&:first-child {
			margin-right: 2%;
		}

		&:last-child {
			margin-right: 2%;
		}
	}

	@media (max-width: $mobile-width) {
		button {
			width: 48%;
		}
	}

	@media (max-width: $half-mobile-width) {
		flex-direction: column;

		button {
			width: 100%;
			margin-right: 0;
			margin-left: 0;
			margin-top: 2%;
		}
	}
}