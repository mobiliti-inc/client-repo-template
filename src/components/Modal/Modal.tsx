import * as React from 'react';
import cx from 'classnames';
import CSSModules from 'react-css-modules';
import Button from '../Button/Button';
import * as styles from './Modal.scss';
// We'll replace this with FontAwesome icons
// import { iconClose } from '../../assets';

export const MODAL_OVERLAYS = {
	GRAY: 'gray',
	WHITE: 'white',
	DARK_GRAY: 'dark-gray'
};

interface ModalProps {
	title?: string;
	subTitle?: string;
	buttonText?: string;
	buttonClassName?: string;
	visible: boolean;
	onButtonClick?: (...args: any[]) => any;
	icon?: React.ReactNode;
	scrolling: boolean;
	plain?: boolean;
	overlay?: string;
	freezeOverlay?: boolean;
	noIcon?: boolean;
	showCloseIcon?: boolean;
	showButtonLoader?: boolean;
	onIconClose: (...args: any[]) => any;
	headerClassName?: string;
	bodyClassName?: string;
	footerClassName?: string;
	plainHeader?: boolean;
	modalParentClass?: string;
	noPointer?: boolean;
}

const Modal: React.FC<ModalProps> = (props) => {
	const {
		title,
		subTitle,
		children,
		buttonText,
		buttonClassName,
		onButtonClick,
		icon,
		scrolling,
		visible,
		plain,
		noIcon,
		overlay,
		showCloseIcon,
		freezeOverlay,
		onIconClose,
		showButtonLoader,
		headerClassName,
		bodyClassName,
		noPointer,
		footerClassName,
		plainHeader,
		modalParentClass
	} = props;

	const [modalVisible, setModalVisible] = React.useState<boolean>(false);

	React.useEffect(() => {
		setModalVisible(visible);
	}, []);

	const updateModalVisibility = (visible: boolean) => setModalVisible(visible);

	React.useEffect(() => {
		updateModalVisibility(visible);
	}, [visible]);

	const onDialogClick = (event: React.MouseEvent<HTMLElement>) => {
		event.stopPropagation();
	};

	const getModalBackgroundName = () => {
		return overlay && Object.values(MODAL_OVERLAYS).indexOf(overlay) > -1 ? `modal--${overlay}` : 'modal--white';
	};

	const getClasses = () => {
		const modalBackgroundName = getModalBackgroundName();
		const modalClass = modalVisible ? 'visible' : 'hide';
		const classes = [modalBackgroundName, modalClass, { 'no-pointer': noPointer }];
		return cx(classes);
	};

	const close = (event: React.MouseEvent<HTMLElement>) => {
		event.preventDefault();
		onIconClose();
		setModalVisible(false);
	};

	const overlayClose = (event: React.MouseEvent<HTMLElement>) => {
		if (onIconClose()) {
			return onIconClose();
		}

		if (freezeOverlay) {
			event.stopPropagation();
		} else {
			close(event);
		}
	};

	const classes = getClasses();

	const modalContentClass = cx('modal-content', { scroll: scrolling, 'modal-plain': plain });
	return (
		<div styleName={classes} role="presentation" onClick={event => overlayClose(event)}>
			<div styleName={modalContentClass} className={modalParentClass} role="presentation" onClick={event => onDialogClick(event)}>
				<div styleName="modal-container" role="presentation">
					{plain ? (
						<React.Fragment>
							{!plainHeader && (
								<div styleName="plain-modal-header" role="presentation" onClick={event => close(event)}>
									{/* <img styleName="modal-icon-close" src={iconClose} alt="Close" /> */}
									X
								</div>
							)}
							{children}
						</React.Fragment>
					) : (
							<React.Fragment>
								<div styleName={cx('header', { 'header-no-icon': noIcon })} className={headerClassName}>
									{showCloseIcon && (
										<div styleName="plain-modal-header" role="presentation" onClick={event => close(event)}>
											{/* <img src={iconClose} alt="Close" /> */}
											X
										</div>
									)}
									{!noIcon && (
										<div styleName="modal-icon">
											{icon && icon}
										</div>
									)}
									<p styleName="modal-title">{title}</p>
									<p styleName="modal-sub-title">{subTitle}</p>
								</div>
								<div styleName="modal-body" className={bodyClassName}>
									{children}
								</div>
								<div styleName="modal-footer" className={footerClassName}>
									<Button customStyles={buttonClassName} onClick={onButtonClick} showLoader={showButtonLoader}>
										{buttonText}
									</Button>
								</div>
							</React.Fragment>
						)}
				</div>
			</div>
		</div>
	);
};

export default CSSModules(Modal, styles, { allowMultiple: true });
