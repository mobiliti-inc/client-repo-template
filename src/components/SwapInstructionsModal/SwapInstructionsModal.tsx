import Modal from "../Modal/Modal";
import Button from "../Button/Button";
import ItemList from "../ItemList/ItemList";
import styles from "./SwapInstructionsModal.scss";

type SwapInstructionsModalProps = {
	modalVisible: boolean,
	scrolling: boolean,
	onCancel: (...args: any[]) => any,
	onContinue: (...args: any[]) => any,
	freezeOverlay: boolean
};

const SwapInstructionsModal: React.SFC<SwapInstructionsModalProps> = ({ modalVisible, scrolling, onContinue, onCancel, freezeOverlay }) => (
	<Modal
		plain
		plainHeader
		visible={modalVisible}
		overlay="gray"
		scrolling={scrolling}
		modalParentClass={styles.modal}
		freezeOverlay={freezeOverlay}
		noPointer
	>
		<p styleName="modal-title">Swap Vehicles</p>
		<div styleName="modal-ctn">
			<ItemList header="Find your new vehicle" subHeader="Browse Mobilitiâ€™s large selection of vehicles from local dealerships." number="1" />
			<ItemList header="Schedule your return" subHeader="Schedule your return for your current vehicle prior to picking up your new ride!" number="2" />
			<ItemList header="Reserve your new vehicle" subHeader="Reserve your new vehicle and set your pick up date. Enjoy the adventure!" number="3" />
		</div>
		<div styleName="button-container">
			<Button onClick={onContinue} ripple>
				Continue
			</Button>
			<Button onClick={onCancel} inverted>
				Cancel
			</Button>
		</div>
	</Modal>
);

export default CSSModules(SwapInstructionsModal, styles, { allowMultiple: true });
