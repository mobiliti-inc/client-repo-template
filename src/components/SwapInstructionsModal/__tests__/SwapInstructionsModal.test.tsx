import SwapInstructionsModal from '../SwapInstructionsModal';
import Modal from '../../../components/Modal/Modal';

describe('SwapInstructionsModal', () => {
	const props = {
		modalVisible: true,
		scrolling: true,
		onContinue: jest.fn(),
		onCancel: jest.fn(),
		freezeOverlay: false,
	};
	const component = (<SwapInstructionsModal {...props} />);

	it('renders', () => {
		expect(component).toMatchSnapshot();
	});

	it('should be a `Modal`', () => {
		const tree = shallow(component);
		expect(tree.find(Modal)).toHaveLength(1);
	});
});
