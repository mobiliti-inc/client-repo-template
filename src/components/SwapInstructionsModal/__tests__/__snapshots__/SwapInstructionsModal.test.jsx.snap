// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapInstructionsModal renders 1`] = `
<SwapInstructionsModal
  freezeOverlay={false}
  modalVisible={true}
  onCancel={[MockFunction]}
  onContinue={[MockFunction]}
  scrolling={true}
/>
`;
